### ISSUE TITLE
GetInstance Fails on Android

### ISSUE THREAD

**Issue by mrbiggred** [realm/realm-dotnet#1421]
## Goals

Looking for help on why calling Realm.GetInstance() on Android crashes but not in iOS and not in debug mode in Android.

## Expected Results

Calling Realm.GetInstance() should return a database to use.

## Actual Results

I get the following exception when calling GetInstance:

05-29 09:56:56.381 20392 20392 I mono-stdout: System.TypeInitializationException: The type initializer for 'Realms.Realm' threw an exception. ---> System.DllNotFoundException: realm-wrappers
05-29 09:56:56.383 20392 20392 I mono-stdout:   at (wrapper managed-to-native) Realms.NativeCommon:register_notify_realm_changed (Realms.NativeCommon/NotifyRealmCallback)
05-29 09:56:56.383 20392 20392 I mono-stdout:   at Realms.Realm..cctor () [0x00017] in <4ea5c2fb68234dddbca0e9c8634b6192>:0
05-29 09:56:56.383 20392 20392 I mono-stdout:    --- End of inner exception stack trace ---

## Steps to Reproduce

Create an Xamarin Forms 2.3.5.239-pre3 application using Prism 6.3.0 and Realm.Database 1.4.0.  The portable project is a .NET standard 1.4 project instead of a PCL.  The application runs fine when launched from Visual Studio, both debug and release, on the actual device (Google Pixel).  The application runs fine when installed using a release ipa file.  The application fails when installed using a release apk file with the above exception.

I had a similar application that with no errors but the portable project was PCL and I used the latest stable version of Xamarin Forms (2.3.4.247).

It appears to be a linking issue but I can't figure it out.  Any help would be much appreciated.  Please let me know if you need more information.

## Code Sample

    internal class FooService: IFooService
    {
        private readonly Realm _realm;

        public FooService()
        {
            try
            {
                _realm = Realm.GetInstance();
            }
            catch (Exception e)
            {
                Console.WriteLine(e);
                throw;
            }
        }
    }

I use TeamCity to build the apk file.  After doing a full rebuild, to build the iPhone ipa, I have a build step that creates the apk that has the following settings:

Runner Type: Visual Studio
Targets: PackageForAndroid
Command line parameters: /p:SolutionDir=Source/Foo/Foo.Android/

## Version of Realm and Tooling

- Visual Studio 2015
- Xamarin 4.5.0.476
- Xamarin.Android 7.3.1.2
- Realm DB 1.4.0
- Xamarin Forms 2.3.5.239-pre3

**Comment by nirinchev** [comment]
It does seem like the final apk is missing the native portion that Realm relies on. Can you try to create the release .apk from VS and see if it exhibits the same issue as the one built from TeamCity?

**Comment by mrbiggred** [comment]
I deleted the bin and obj folders then right clicked on the Android project in Visual Studio and choose deploy.  Everything ran fine.  I did this for both debug and release modes with the Any CPU target.

I then tried building the using the command line on my workstation that mimicked the build machine.  It looked like:

msbuild /p:configuration=Release /t:Rebuild;PackageForAndroid /p:SolutionDir=C:\Repos\Foo\Foo.Android\ 

I then signed it using the my debug key and it installed it using adb via the command line.  It ran fine.  I can't figure out what is different between my development and build machine.  Any suggestions?

**Comment by nirinchev** [comment]
I'm not familiar with TeamCity, but could you try to do a full rebuild for Android as well?

**Comment by mrbiggred** [comment]
I figured out the issues or at least a workaround.  The problem only occurs if I us the TeamCity MSBuild Runner.  If I call the build from the command line then there is no issues.  I changed the TeamCity build from the MSBuild Runner to the Command Line Runner and I no longer get the TypeInitializationException.  The command line I use is:

"C:\Program Files (x86)\MSBuild\14.0\Bin\msbuild.exe" /p:configuration=Release /t:Rebuild;PackageForAndroid /p:SolutionDir=%teamcity.build.checkoutDir%/Source/Foo/Foo.Android/

It appears the issue has something to do with a difference between TeamCity MSBuild and the Command Line MSBuild.  If I get bored I might review the output from both builds but for now I'll just leave it be.

Thank you very much for you help and patience.  I've only been using Realm Database for a month but so far I've really enjoyed working with it.  Keep up the good work.
